chore(API docs): update header of angular-patternfly API Docs site

dabeng opened the pull request:
## Description
This is the sub-task of [PTNFLY-2467](https://patternfly.atlassian.net/browse/PTNFLY-2467).

@dlabrecq @dtaylor113 @jeff-phillips-18 @bleathem @LHinson 
Based on the PatternFly Showcase Design[1], I've created a PR for updating the header.
The demo link is here: https://rawgit.com/dabeng/angular-patternfly/new-header-dist/docs/index.html#/api

dtaylor113 commented:
Hi, top left links seem to be cut off:

![image](https://user-images.githubusercontent.com/12733153/29072732-ef329176-7c16-11e7-86fd-e54078d06f0b.png)

Also, the "Components" and "Contribute" links don't seem to go anywhere and do not change the 'selected' state from "Getting Started"

![image](https://user-images.githubusercontent.com/12733153/29072792-397b98d6-7c17-11e7-83dc-2454c992ab38.png)

This PR does not seem complete, customer ready.  I'd be hesitant to merge this.  Can we add a '[WIP]' to the title of this PR and consider not merging into master but  rather to a dev branch until everything is ready?

Thanks,
Dave

jeff-phillips-18 commented on Gruntfile.js:
```diff
@@ -194,6 +194,7 @@ module.exports = function (grunt) {
           styles: ['node_modules/datatables.net-dt/css/jquery.dataTables.css',
             'node_modules/patternfly/dist/css/patternfly.css',
             'node_modules/patternfly/dist/css/patternfly-additions.css',
+            'http://rawgit.com/junezhang/showcase/master/dist/styles/patternfly-showcase.css',
```
Is there a patternfly based repo this could link to rather than an individuals repo?

LHinson replied:
+1 

Each repo should have it's own get started and contribute pages, specific with details and information for that particular repo. June has already populated the one for Angular 1 here: http://rawgit.com/junezhang/showcase/master/index.html

@jeff-phillips-18 @dtaylor113 does this information look up to date and correct? If so, @dabeng can we introduce this to the angular repo with this PR?

jeff-phillips-18 replied:
The README file needs some work which will then need to be incorporated into this. :(
I am working on updating it.

dtaylor113 commented:
> does this information look up to date and correct? If so, @dabeng can we introduce this to the angular repo with this PR?

Hi @LHinson, I'm not sure what you are asking?  If this PR is merged it will go into the angular 1 repo, but IMO it is not ready to be merged (pls see my comments above).

dtaylor113 commented:
I'm not sure how this PR works with PF Showcase?  PF Showcase looks like this:

![image](https://user-images.githubusercontent.com/12733153/29074258-981cf920-7c1c-11e7-9ff5-4fc468232d21.png)

Whereas this PR/ngDoc looks like this:

![image](https://user-images.githubusercontent.com/12733153/29074276-b034fbca-7c1c-11e7-9cca-e64512aeadb4.png)

Other than the CSS issues which @dabeng mentions, how do these integrate together?  Are they supposed to look exactly the same?  PF Showcase masthead title says "PatternFly Showcase", this PR/ngDoc masthead title is "PATTERNFLY ANGUALR"  -not the same case/caps

dabeng commented:
@dtaylor113 , maybe you need the following processing.

<img width="1600" alt="screen shot 2017-08-09 at 11 35 34 am" src="https://user-images.githubusercontent.com/5689075/29104266-1aa9a042-7c8a-11e7-83ab-0780bd042e35.png">

dtaylor113 commented:
@dabeng , thanks!  That did the trick!  

> Currently, I linked an external css file[2], do we need to add an internal css file in this repo?

There are less files [here](https://github.com/patternfly/angular-patternfly/tree/master/styles) for angular-patternfly, you can add your styles to these, or create another less file.  It should be included in this PR.

Couple of questions/issues:

1. The [page](http://rawgit.com/junezhang/showcase/master/index.html) June has created, is that "Getting Started" or "Contribute"?
2. The ngDocs page, is that going to be "Components"?
3. IMO all the masthead links need to goto appropriate pages before this PR can be merged.

Thanks!

dtaylor113 commented:
@dabeng ,  extra credit if you can solve the 'ANGULAR Error' for the tab title :-)

![error](https://user-images.githubusercontent.com/12733153/29122620-fe9df1bc-7ce0-11e7-8866-f9576aff78ac.png)

dtaylor113 commented:
Hi @dabeng, we just merged #554 which does automatic semantic releasing for angular-patternfly.  You'll need to redo this PR's commit message to this [format](https://github.com/semantic-release/semantic-release#default-commit-message-format).  -thanks

dabeng commented:
Hi @dtaylor113 , thanks a lot for your improvement advices :)

1. I have created showcase.less under "styles" folder of angular-patternfly;
2. The page June has created is  "Getting Started" page.  
3. You are right. The API Docs is going to be "Components" page;
4. I have updated the masthead links. While , standalone get-stared and contribute pages of angular-patternfly is necessary;
5. IMO, I think it's also necessary to remove the "#" from the window.location;
6. I think "extra credit" belongs to another scope, we can fix it outside this PR.

dtaylor113 commented:
Hi @dabeng, 
Ok, thanks for the update.
I see you updated the PR title/subject to include the semantic release format: "feat(API docs)", unfortunately this needs to be the commit message, not the PR title.   You will need to do a 'git rebase -i' to change the commit message.  -thanks!

dabeng commented:
Hi @dtaylor113 , commit message is modified accordingly.

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,98 @@
+            <span class="icon-bar"></span>
+          </button>
+          <div class="navbar-brand">
+            <a id="logo" title="PatternFly Angular" href="#">
```
This should be 'Angular Patternfly'

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,98 @@
+        <div class="navbar-collapse navbar-collapse-pf-site collapse">
+          <ul id="menu-primary" class="nav navbar-nav navbar-right">
+            <li>
+              <a href="get-startted">Get Started</a>
```
Misspelling. This is a broken link.

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,98 @@
+              <a href="#">Components</a>
+            </li>
+            <li>
+              <a href="contribute">Contribute</a>
```
This is a broken link.

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,98 @@
+                  <a href="https://rawgit.com/patternfly/patternfly-react/gh-pages/index.html">PatternFly React</a>
+                </li>
+                <li>
+                  <a href="http://www.patternfly.org/patternfly-css/">PatternFly CSS</a>
```
I'm not sure this is the correct link. It takes you to an inner patternfly org page which isn't formatted for stand alone.

dabeng replied:
This link is the only valid link for patternfly css.

jeff-phillips-18 replied:
That page doesn't look ready for prime time. Do we have an estimate on when it will be?

junezhang replied:
@jeff-phillips-18 sorry, not catch your point about the prime time.
This showcase list is a window for user getting to know what we are addressing in the framework implementation like the below picture. 
<img width="760" alt="screen shot 2017-08-22 at 3 13 21 pm" src="https://user-images.githubusercontent.com/701009/29553366-73d8ac80-874d-11e7-8843-71ed071abd11.png">

Even though some showcase projects are pretty new, like PatternFly-React, PatternFly-WebComponents, we also want user to see the progress, and welcome any contribution. How do you think about it?  also here at @LHinson , want to hear any suggestions?

LHinson replied:
+1 with @jeff-phillips-18. The CSS should be linked in the future but we aren't ready yet. It should be removed for now and we can add at a later date.

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,98 @@
+          <a href="">Menu</a>
+          <ul>
+            <li>
+              <a href="get-startted">Get Started</a>
```
Misspelling

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,98 @@
+            </li>
+            <li class="dropdown">
+              <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+                <i class="fa fa-th-large" aria-hidden="true"></i>
```
aria-hidden=true and should have an sr-only span for "Patternfly Showcases"

jeff-phillips-18 commented on Gruntfile.js:
```diff
@@ -163,7 +164,7 @@ module.exports = function (grunt) {
       ngdocs: {
         options: {
-          title: 'Angular Patternfly Documentation',
+          title: 'ANGULAR',
```
This should be 'Angular Patternfly'

dabeng replied:
Sorry Jeff, if do that, the heading will become "Patternfly - Angular Patternfly", Don't you feel it will be wired?

jeff-phillips-18 replied:
It should end up as simply "Angular Patternfly", whatever that takes.

jeff-phillips-18 commented:
the showcase.less file should be in misc (which is not less ready) not in style. Putting it in styles will add it to the distribution and these styles should NOT be in the distribution.

jeff-phillips-18 commented:
Not sure that I believe fixing:
![image](https://user-images.githubusercontent.com/11633780/29375565-757b857c-8283-11e7-99d3-fefdc3bbc9dc.png)

is extra credit. It seems caused by this PR.

dtaylor113 commented:
@jeff-phillips-18, unfortunately not.  I see "Error: Page Not Found" in our published [ngDocs](https://www.patternfly.org/angular-patternfly/#/api)

dabeng commented:
Hi @jeff-phillips-18 , I have moved showcase css code to folder misc and updated the details you mentioned above :blush:

jeff-phillips-18 commented:
Also, I don't believe this is a feat. No reason to bump the release for this. IMO, this is a chore since there are not distribution changes.

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,88 @@
+                  <a href="https://patternfly-webcomponents.github.io/components.html">PatternFly-WebComponents</a>
+                </li>
+                <li>
+                  <a href="https://rawgit.com/patternfly/patternfly-react/gh-pages/index.html">PatternFly React</a>
```
I see different formatting for the names

- Patternfly-ng
- Patternfly-WebComponents
- Patternfly React
- Patternfly CSS

These should either be the repo name:

- patternfly-ng
- patternfly-webcomponents
- patternfly-react
- patternfly-css

Or the titles:

- Patternfly NG
- Patternfly Web Components
- Patternfly React
- Patternfly CSS

LHinson commented:
Add this comment inline above but will add here too... 

The PatternFly PatternLab CSS should be linked in the future but we aren't ready yet. It should be removed for now and we can add at a later date. This is for the CSS.Next initiative.

dabeng commented:
@dtaylor113 , please review again.

dtaylor113 commented:
Hi @dabeng, I notice a weird 'white-out' effect when I move the mouse pointer off of the app launcher icon:

![image](https://user-images.githubusercontent.com/12733153/29618523-71c6539a-87e6-11e7-997b-1e239f987f06.png)
![image](https://user-images.githubusercontent.com/12733153/29618530-75678686-87e6-11e7-81b8-fdcd7211ba44.png)

dtaylor113 commented:
Hi @dabeng, I don't know if GitHub is acting up at the moment but none of the links in the app. launcher seem to work:
1. patternfly-ng hangs with "Loading..."
2. web-components does go to another page, but masthead is white?  
![image](https://user-images.githubusercontent.com/12733153/29618656-e809e24c-87e6-11e7-9632-14476e78e6e0.png)
3. patternfly-react results in "No server is currently available to serve your request".

Again, I don't know if GitHub is acting up or if these landing pages aren't quite done yet.

junezhang commented:
@LHinson @jeff-phillips-18 Okay, got it. I will remove the css part from the list.

junezhang commented:
@dtaylor113 Thanks for point the "white-out" bug, I've fixed this bug on this link http://rawgit.com/junezhang/showcase/master/index.html 
@dabeng please help to update. Thanks.

dabeng commented:
Hi @dtaylor113 @jeff-phillips-18 , I have updated PR according to your advices. Please review.

FYI, @dtaylor113 , regarding external links, I got the following snapshot, they looks great. Maybe the issue you mentioned above stems from internet or network agent.

FYI, @jeff-phillips-18 , I have removed all unready links from the navigation bar.

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,82 @@
+          </button>
+          <div class="navbar-brand">
+            <a id="logo" title="Angular Patternfly" href="#">
+              PATTERNFLY-<span class="secondary-logo" ng-bind-template="<%= title %>"></span>
```
Do we need the all caps PATTERNFLY here? Is this the design across all the showcases?

dabeng replied:
Yes @jeff-phillips-18 , this is the uniformed design across all the showcases.

jeff-phillips-18 replied:
So are other components PATTERNFLY - Patternfly NG, and PATTERNFLY - Patternfly Web Components? Seems redundant.

junezhang replied:
@jeff-phillips-18 It should be PATTERNFLY-NG, PATTERNFLY-REACT, PATTERNFLY-WEB COMPONENTS, PATTERNFLY-ANGULAR.  so here <%= title %> the title should be ANGULAR.

jeff-phillips-18 replied:
This repository has been around far too long to start calling it PATTERNFLY-ANGULAR. It has been known as Angular Patternfly and should remain that way.

junezhang replied:
@jeff-phillips-18 yes, we discussed it before. sorry for forgetting it.
so here the name should be ANGULAR PATTERNFLY, does it make sense?

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,82 @@
+            <li class="dropdown">
+              <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+                <i class="fa fa-th-large" aria-hidden="true">
+                  <span class="sr-only">PatternFly Showcases</span>
```
This should use pfApplicationLauncher.

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,82 @@
+                  <a href="https://rawgit.com/patternfly/patternfly-ng/master-dist/dist-demo/#/">PatternFly NG</a>
+                </li>
+                <li>
+                  <a href="https://patternfly-webcomponents.github.io/components.html">PatternFly WebComponents</a>
```
WebComponents should be "Web Components" to match the page it directs you to. The page it directs you to is titled simply "Web Components". Not sure who is responsible, but I would expect that page's title to be "_Patternfly_ Web Components"

jeff-phillips-18 commented on grunt-ngdocs-index.tmpl:
```diff
@@ -89,30 +89,82 @@
+                  <a href="https://rawgit.com/patternfly/patternfly-ng/master-dist/dist-demo/#/">PatternFly NG</a>
+                </li>
+                <li>
+                  <a href="https://patternfly-webcomponents.github.io/components.html">PatternFly WebComponents</a>
```
WebComponents should be "Web Components"

dabeng commented:
Sorry @jeff-phillips-18 , I think I couldn't accept your advice to replace plain "dropdown-menu" with pfApplicationLauncher because there are too much UI differences between them. Much efforts must be made to modify the existing pfApplicationLauncher to abide by the uniformed header style designed by june, even I need to refactor the  directive pfApplicationLauncher.

jeff-phillips-18 commented:
You would just use the patternfly version, not necessarily the Angular directive. I'm not sure you could get our directives to work at this point in the ng-docs template. 

You could try if you like by adding a file in the misc directory as a controller for the navbar and put those settings in there (including this new js file in the 'scripts' portion of the ngdocs settings in the Gruntfile). Then setting the controller for the nav and using the directive. It would be interesting to get this working as it could lead to further changes to the page and more that we could control.

Changing to the simple Patternfly version of the application launcher is probably plenty though as there isn't much need here for any additional logic in the controller. 

However, there may be some adjustments necessary as I'm not sure how well the application launcher will work outside of a PF horizontal or vertical nav bar. It should so you may need to file some issues and do some workarounds until it is fixed. This is a good test for its adaptability 😉

dabeng commented:
Hi @jeff-phillips-18 , I have applied patternfly application launcher in the new header.

jeff-phillips-18 commented:
![image](https://user-images.githubusercontent.com/11633780/29817167-eed66b52-8c84-11e7-819a-94910822d4a5.png)

This is what I meant by needing some adjustments. The dropdown list is too far right (partially off screen), the arrow should be centered on the trigger.

junezhang commented on misc/showcase.css:
```diff
@@ -0,0 +1,740 @@
+@media (min-width: 768px) {
+  .showcase header .navbar-utility .applauncher-pf .dropdown-menu {
+    margin-top: 0;
+  }
```
@dabeng you miss one piece of css code here

@media (min-width: 768px) {
.showcase header .navbar-utility .applauncher-pf .dropdown-menu {
    margin-top: 0;
    right: 5px;
}
}

junezhang commented on misc/showcase.css:
```diff
@@ -0,0 +1,740 @@
+    margin-top: 0;
+  }
+}
+.showcase header .dropdown > a:link,
```
@dabeng you missed one piece of css code here

.dropdown-kebab-pf .dropdown-menu.dropdown-menu-right::after,
.dropdown-kebab-pf .dropdown-menu.dropdown-menu-right::before {
right: 10px;
}

junezhang commented on misc/showcase.css:
```diff
@@ -0,0 +1,740 @@
+  .showcase header .navbar-utility .applauncher-pf .dropdown-menu {
+    margin-top: 0;
+  }
+}
```
@dabeng you missed one piece of css code here

.dropdown-kebab-pf .dropdown-menu.dropdown-menu-right::after,
.dropdown-kebab-pf .dropdown-menu.dropdown-menu-right::before {
right: 10px;
}

dabeng commented:
Hi @jeff-phillips-18 @junezhang , I have updated showcase.css and got the following snapshots:
<img width="1600" alt="screen shot 2017-08-31 at 12 28 49 pm" src="https://user-images.githubusercontent.com/5689075/29906652-e6c3c9c0-8ddb-11e7-912d-6a586cb81bad.png">
<img width="1600" alt="screen shot 2017-08-31 at 12 29 01 pm" src="https://user-images.githubusercontent.com/5689075/29906653-ead7cc46-8ddb-11e7-9b07-9ffa270cbeec.png">
<img width="1600" alt="screen shot 2017-08-31 at 12 29 07 pm" src="https://user-images.githubusercontent.com/5689075/29906657-eeab67d8-8ddb-11e7-8a78-89f91f8177ed.png">
<img width="1600" alt="screen shot 2017-08-31 at 12 29 16 pm" src="https://user-images.githubusercontent.com/5689075/29906660-f281b84e-8ddb-11e7-9cd0-9dc0e00c5d50.png">

jeff-phillips-18 commented:
I'm not seeing this fix when I build/run.

dtaylor113 commented:
Sorry, what are the differences between the screenshots?

junezhang commented:
@dtaylor113 I think the difference between the screenshots is in different browsers, what they are looking like.

junezhang commented:
yes, remembered it, we discussed it before. so we will use ANGLUAR PATTERNFLY. does it make sense?




June







On Thu, Aug 31, 2017 at 8:43 PM +0800, "Jeffrey Phillips" <notifications@github.com> wrote:












@jeff-phillips-18 commented on this pull request.





In grunt-ngdocs-index.tmpl:
> -              <img class="pull-left" src="<%= image %>">
-            <% if (imageLink) { %>
+<body class="showcase">
+  <header>
+    <nav class="navbar navbar-default navbar-fixed-top">
+      <div class="container-fluid">
+        <div class="navbar-header">
+          <button type="button" class="navbar-toggle" ng-click="isOpen = !isOpen">
+            <span class="sr-only">Toggle navigation</span>
+            <span class="icon-bar"></span>
+            <span class="icon-bar"></span>
+            <span class="icon-bar"></span>
+          </button>
+          <div class="navbar-brand">
+            <a id="logo" title="Angular Patternfly" href="#">
+              PATTERNFLY-<span class="secondary-logo" ng-bind-template="<%= title %>"></span>



This repository has been around far too long to start calling it PATTERNFLY-ANGULAR. It has been known as Angular Patternfly and should remain that way.



—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

dabeng commented:
@jeff-phillips-18 , maybe you can clean up your browser cache or directly review the build/run results on this address -- https://rawgit.com/dabeng/angular-patternfly/new-header-dist/docs/index.html#/api

dabeng commented:
Hi @dtaylor113 , I listed snapshots in order to demonstrate that patternfly application launcher menu has the same styles in different browers 😊

dtaylor113 commented:
> Hi @dtaylor113 , I listed snapshots in order to demonstrate that patternfly application launcher menu has the same styles in different browers

Hi @dabeng, ok, thanks for the explanation.
